chainer/link.py

```diff
@@ -171,28 +239,21 @@ def add_param(self, name, shape=None, dtype=numpy.float32,
                 initializer. Note that in these cases, ``dtype`` argument is
                 ignored.
 
+        Returns:
+            ~chainer.Parameter: The registered parameter object.
```

Comment:
Why do you change the specification of `add_param` to return the parameter to be registered. As `add_param` is deprecated, I thought we do not have to do that. And actually no code in this PR assumes that ` add_param` returns an instance of `Parameter`.